tests/Browser/PilotPassportChallengeTest.php

```diff
@@ -0,0 +1,186 @@
+use App\PilotPassportEnrollment;
+use App\PilotPassportLog;
+use App\RealopsPilot;
+use Config;
+use Laravel\Dusk\Browser;
+use Tests\DuskTestCase;
+
+class PilotPassportChallengeTest extends DuskTestCase {
+
+    const TEST_AIRPORT_ID = 'GSO';
+    const TEST_CHALLENGE_ID = 1;
+
+    public function test_home_public(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->assertSee('What is the ZTL Pilot Passport Challenge?');
+        });
+    }
+
+    public function test_home_authenticated(): void {
+        $this->loginSamplePilot();
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->assertSee('Welcome, ');
+        });
+    }
+    
+    public function test_enrollments_view(): void {
+        $this->loginSamplePilot();
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Enrollments')
+                    ->assertSee('Select a path below and enroll to get started.');
+        });
+    }
+
+    public function test_enroll(): void {
+        $this->loginSamplePilot();
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+            ->clickLink('Enrollments')
+            ->press('@enroll_1')
+            ->assertSee('You are now enrolled in the ZTL Pilot Passport program!');
+        });
+    }
+
+    public function test_passport_book_view(): void {
+        $this->loginSamplePilot(true);
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Passport Book')
+                    ->assertSee('Visited on: ');
+        });
+    }
+
+    public function test_achievements_view(): void {
+        $this->loginSamplePilot(true);
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Achievements')
+                    ->assertSee('Completed on: ');
+        });
+    }
+
+    public function test_settings_view(): void {
+        $this->loginSamplePilot();
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->assertSee('ZTL understands that some pilots may want to participate');
+        });
+    }
+
+    public function test_change_privacy_settings(): void {
+        $this->loginSamplePilot();
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->radio('@privacy', '2')
+                    ->press('Save Settings')
+                    ->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->assertRadioSelected('@privacy', '2');
+        });
+    }
```

Comment:
This is another good spot to make sure this has been saved in the backend

Reply:
Done!